crates/bounded_planet_core/src/math/screen_transformations.rs

```diff
@@ -0,0 +1,24 @@
+use bevy::prelude::*;
+
+/// Convert a screen point into a NDC (Normalized Device Coordinate) point.
+/// 
+/// Uses screen size as part of the conversion math.
+pub fn screen_to_ndc(point: Vec2, screen_size: Vec2) -> Vec3 {
+    ((point / screen_size) * 2.0 - Vec2::new(1.0, 1.0)).extend(0.0)
+}
+
+/// Convert a NDC (Normalized Device Coordinate) point into a world point.
+/// 
+/// Uses the camera transform matrix (preferably global) and the camera projection matrix as part of the conversion math.
+pub fn ndc_to_world(point: Vec3, camera_transform: Mat4, projection_matrix: Mat4) -> Vec3 {
```

Comment:
...and then accept that `NDC` wrapper here instead.

Reply:
That's not a bad idea, but is also possibly a bit overzealous...

I think I'll leave it for now, and have whoever upgrades some math stuff next take a look at it.